packages/lodestar/src/chain/validation/aggregateAndProof.ts

```diff
@@ -24,126 +22,124 @@ export async function validateGossipAggregateAndProof(
-  } catch (error) {
-    throw new AttestationError({
-      code: AttestationErrorCode.AGGREGATOR_NOT_IN_COMMITTEE,
+  // [REJECT] aggregate_and_proof.selection_proof selects the validator as an aggregator for the slot
+  // -- i.e. is_aggregator(state, aggregate.data.slot, aggregate.data.index, aggregate_and_proof.selection_proof) returns True.
+  if (!isAggregatorFromCommitteeLength(config, committee.length, aggregateAndProof.selectionProof)) {
+    throw new AttestationGossipError(GossipAction.REJECT, {
+      code: AttestationErrorCode.INVALID_AGGREGATOR,
       job: attestationJob,
     });
   }
 
-  if (!committee.includes(aggregateAndProof.message.aggregatorIndex)) {
-    throw new AttestationError({
+  // [REJECT] The aggregator's validator index is within the committee
+  // -- i.e. aggregate_and_proof.aggregator_index in get_beacon_committee(state, aggregate.data.slot, aggregate.data.index).
+  if (!committee.includes(aggregateAndProof.aggregatorIndex)) {
+    throw new AttestationGossipError(GossipAction.REJECT, {
       code: AttestationErrorCode.AGGREGATOR_NOT_IN_COMMITTEE,
       job: attestationJob,
     });
   }
 
-  if (!isAggregatorFromCommitteeLength(config, committee.length, aggregateAndProof.message.selectionProof)) {
-    throw new AttestationError({
-      code: AttestationErrorCode.INVALID_AGGREGATOR,
+  // [IGNORE] The block being voted for (aggregate.data.beacon_block_root) has been seen
+  // (via both gossip and non-gossip sources) (a client MAY queue aggregates for processing once block is retrieved).
+
+  // [REJECT] The block being voted for (aggregate.data.beacon_block_root) passes validation.
+  if (await db.badBlock.has(aggregate.data.beaconBlockRoot.valueOf() as Uint8Array)) {
+    throw new AttestationGossipError(GossipAction.REJECT, {
+      code: AttestationErrorCode.KNOWN_BAD_BLOCK,
       job: attestationJob,
     });
   }
 
-  const slot = attestation.data.slot;
-  const epoch = computeEpochAtSlot(config, slot);
-  const indexedAttestation = attestationTargetState.getIndexedAttestation(attestation);
-  const aggregator = attestationTargetState.index2pubkey[aggregateAndProof.message.aggregatorIndex];
+  // [REJECT] The current finalized_checkpoint is an ancestor of the block defined by aggregate.data.beacon_block_root
+  // -- i.e. get_ancestor(store, aggregate.data.beacon_block_root, compute_start_slot_at_epoch(store.finalized_checkpoint.epoch)) == store.finalized_checkpoint.root
 
+  // [REJECT] The aggregate_and_proof.selection_proof is a valid signature of the aggregate.data.slot
+  // by the validator with index aggregate_and_proof.aggregator_index.
+  // [REJECT] The aggregator signature, signed_aggregate_and_proof.signature, is valid.
+  // [REJECT] The signature of aggregate is valid.
+  const slot = aggregate.data.slot;
+  const epoch = computeEpochAtSlot(config, slot);
+  const aggregator = attestationTargetState.index2pubkey[aggregateAndProof.aggregatorIndex];
```

Comment:
similarly maybe assign a separate `aggregatorIndex` variable to avoid duplicate `aggregateAndProof.aggregatorIndex`

not sure how significant it is but for TreeBacked, we store everything in byte array and a simple `aggregateAndProof.aggregatorIndex` requires us to extract byte array to number, similarly `state.slot` or `state.currentFinalizedCheckpoint.epoch` etc.